File: 0917-reverse-only-letters/0917-reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        s = list(s)  # Convert the string to a list for mutability
        left, right = 0, len(s) - 1
        
        while left < right:
            # Move the left pointer until it points to a letter
            if not s[left].isalpha():
                left += 1
            # Move the right pointer until it points to a letter
            elif not s[right].isalpha():
                right -= 1
            else:
                # Swap the letters at left and right pointers
                s[left], s[right] = s[right], s[left]
                left += 1
                right -= 1
        
        return ''.join(s)  # Convert the list back to a string

File: Difficulty/A. Easy/0917-reverse-only-letters/0917-reverse-only-letters.c
#include <stdio.h>
#include <ctype.h>  // For isalpha()
#include <string.h> // For strlen()

char* reverseOnlyLetters(char* s) {
    int left = 0;
    int right = strlen(s) - 1;

    // Two-pointer approach
    while (left < right) {
        // Move left pointer until it points to a letter
        if (!isalpha(s[left])) {
            left++;
        }
        // Move right pointer until it points to a letter
        else if (!isalpha(s[right])) {
            right--;
        }
        // If both left and right pointers are at letters, swap them
        else {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }

    return s; // Return the modified string
}